File: src/main/java/com/djdapz/stubhub/domain/ProcessedListing.java
package com.djdapz.stubhub.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Value;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;

import static lombok.AccessLevel.PRIVATE;

@FieldDefaults(level = PRIVATE, makeFinal = true)
@Getter
@AllArgsConstructor
@Value
public class ProcessedListing {
    Integer eventId;
    Integer listingId;
    LocalDateTime asOfDate;
    Price currentPrice;
    Price listingPrice;
    Integer sectionId;
    String row;
    Integer quantity;
    String sellerSectionName;
    String sectionName;
    String seatNumbers;
    Integer zoneId;
    String zoneName;
    Integer isGA;
    Boolean dirtyTicketInd;
    String splitOption;
    String ticketSplit;
    Integer score;


    public ProcessedListing(Listing listing, LocalDateTime time, Integer eventId) {
        this.eventId = eventId;
        this.listingId = listing.getListingId();
        this.currentPrice = listing.getCurrentPrice();
        this.listingPrice = listing.getListingPrice();
        this.sectionId = listing.getSectionId();
        this.row = listing.getRow();
        this.quantity = listing.getQuantity();
        this.sellerSectionName = listing.getSellerSectionName();
        this.sectionName = listing.getSectionName();
        this.seatNumbers = listing.getSeatNumbers();
        this.zoneId = listing.getZoneId();
        this.zoneName = listing.getZoneName();
        this.isGA = listing.getIsGA();
        this.dirtyTicketInd = listing.getDirtyTicketInd();
        this.splitOption = listing.getSplitOption();
        this.ticketSplit = listing.getTicketSplit();
        this.score = listing.getScore();
        this.asOfDate = time;
    }
}

File: src/main/java/com/djdapz/stubhub/domain/ProcessedListing.kt
package com.djdapz.stubhub.domain

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime


data class ProcessedListing(
        val asOfDate: LocalDateTime,
        val listingId: Int,
        val eventId: Int,
        val currentPrice: Price,
        val dirtyTicketInd: Boolean,
        val isGa: Int,
        val listingPrice: Price,
        val quantity: Int,
        val row: String,
        val score: Int,
        val seatNumbers: String?,
        val sectionId: Int,
        val sectionName: String,
        val sellerSectionName: String,
        val splitOption: String,
        val ticketSplit: String,
        val zoneId: Int,
        val zoneName: String
) {
    constructor(stubhubListing: StubhubListing, asOfDate: LocalDateTime, eventId: Int) : this(
            asOfDate = asOfDate,
            listingId = stubhubListing.listingId,
            eventId = eventId,
            currentPrice = stubhubListing.currentPrice,
            dirtyTicketInd = stubhubListing.dirtyTicketInd,
            isGa = stubhubListing.isGA,// == 1,
            listingPrice = stubhubListing.listingPrice,
            quantity = stubhubListing.quantity,
            row = stubhubListing.row,
            score = stubhubListing.score,
            seatNumbers = stubhubListing.seatNumbers,
            sectionId = stubhubListing.sectionId,
            sectionName = stubhubListing.sectionName,
            sellerSectionName = stubhubListing.sellerSectionName,
            splitOption = stubhubListing.splitOption,
            ticketSplit = stubhubListing.ticketSplit,
            zoneId = stubhubListing.zoneId,
            zoneName = stubhubListing.zoneName
    )

    @Suppress("unused")
    fun getIsGa() = isGa
}




